### ISSUE TITLE
The SFT optimizer swap still OOMs — #30's mitigation is insufficient

### ISSUE THREAD

**Issue by MatthewLacerda2** [MatthewLacerda2/TinyRefinementModel#189]
When the #157 run flipped into SFT on 2026-08-15, recreating the optimizer OOM'd and killed the process.

## The mitigation is present and did not work

`trainer.py` carries the fix from #30, with the reasoning in a comment:

```python
# OOM-critical ordering (#30): pull the optimizer moments to host, then free the
# old optimizer IN THIS SCOPE, before nnx.Optimizer eagerly allocates the new
# full-size mu/nu on the GPU — otherwise both states coexist for an instant, a
# ~2x spike that OOM'd the 6GB card.
old_state = jax.device_get(nnx.state(optimizer))
del optimizer
gc.collect()
optimizer = create_sft_optimizer(model, old_state)
```

It still failed, on a **572 MB** allocation — parameter-tree sized.

## Two candidate causes, neither confirmed

1. `create_sft_optimizer` allocates the new `mu` (bf16, 265MB) and `nu` (f32, 529MB) **and** has to transfer `old_state` back onto the device to seed them, so an optimizer-sized pair is briefly co-resident anyway. Moving the moments to host defers the spike rather than removing it.
2. Under `cuda_async` (#162, new since #30 was written) the `del` may not return memory to the pool promptly. The #30 fix was designed against BFC's arena.

Distinguishing them needs the card.

## Why this is not urgent

Dormant since #182 — the plateau can no longer trigger a flip, and `SFT_ON_PLATEAU` defaults off. Nothing reaches this code path on a pretrain now.

## Why it still matters

If SFT is ever *wanted*, it is broken, and it breaks in the most expensive way available: mid-run, after a long pretrain, taking the process with it. Also worth noting the same swap exists on the resume path in `trm/train/start.py`, which does **not** even do the `device_get` — it keeps `old_state` on device — so that path is likely worse.

## Lane

`cpu` to reason about and restructure; `gpu` to confirm, and it can only be confirmed by actually performing a swap on the card. A cheap harness that builds the optimizer, swaps it, and reports peak would answer it in minutes — and is the kind of thing the fit-gate (#168) could host.

Found from #157.

**Comment by MatthewLacerda2** [comment]
superseded-by #372: the in-run SFT flip, and the optimizer swap that OOM'd, were removed (#323). No code recreates the optimizer mid-run anymore (`trainer.py` builds it once). If SFT comes back as its own run, it starts from a checkpoint with a fresh optimizer, so this path doesn't exist.
